packages/gateway/src/routes/patientSearch/handler.ts

```diff
@@ -0,0 +1,172 @@
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as Joi from 'joi'
+import * as Hapi from '@hapi/hapi'
+import { fetchFromHearth } from '@gateway/features/fhir/utils'
+
+interface ISearchParams {
+  firstName?: string
+  lastName?: string
+  dateOfBirth?: string
+  dateOfDeath?: string
+  gender?: string
+  offset?: number
+  limit?: number
+}
+
+interface IAttributeError {
+  code?: number
+  message?: string
+}
+
+interface IResponseAttribute {
+  firstName?: string
+  lastName?: string
+  dateOfBirth?: string
+  dateOfDeath?: string
+  gender?: string
+}
+
+const validAttributes = [
+  'firstName',
+  'lastName',
+  'dateOfBirth',
+  'dateOfDeath',
+  'gender'
+]
+
+export async function getPersonsRecord(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+) {
+  const searchParams = request.query as ISearchParams
+  const params = new URLSearchParams()
+  const {
+    firstName,
+    lastName,
+    dateOfBirth,
+    dateOfDeath,
+    gender,
+    limit,
+    offset
+  } = searchParams
+
+  if (firstName) params.set('given', firstName)
+  if (lastName) params.set('family', lastName)
+  if (limit) params.set('_count', String(limit))
+  if (offset) params.set('_getpagesoffset', String(offset))
+  if (dateOfBirth) params.set('dateOfBirth', dateOfBirth)
+  if (dateOfDeath) params.set('dateOfDeath', dateOfDeath)
+  if (gender) params.set('gender', gender)
+
+  const queryString = params.toString()
+  const names = (request.query.names || []) as Array<keyof IResponseAttribute>
+
+  try {
+    const response = (await fetchFromHearth(
+      `/Patient?${queryString}`
+    )) as fhir.Bundle
+    const result = response?.entry?.map((person) => {
+      const { name, gender, birthDate, deceasedDateTime, identifier } =
+        person.resource as fhir.Patient
+
+      const allAttributes: IResponseAttribute = {
+        firstName: name?.[0]?.given?.[0],
+        lastName: name?.[0]?.family?.[0],
+        gender,
+        dateOfBirth: birthDate,
+        dateOfDeath: deceasedDateTime
+      }
+
+      if (names.length > 0) {
+        const requestedAttributes: IResponseAttribute | IAttributeError = {}
+        for (const attr of names) {
+          if (validAttributes.includes(attr)) {
+            requestedAttributes[attr] = allAttributes[attr] || ''
+          } else {
+            requestedAttributes[attr] = {
+              code: '1023',
+              message: 'Unknown attribute name'
+            }
+          }
+        }
+        return requestedAttributes
+      }
+      //this need to be replaced by uin later on
+      const nidIdentifier = identifier?.find((id) => id.type === 'NATIONAL_ID')
```

Comment:
I think we can now change this to be `OSIA_UIN_VID_NID` as it's merged to osia-hackathon PR. The NATIONAL_ID and OSIA_UIN_VID_NID values will be the same though in the Patient identifier.